hibernate-core/src/main/java/org/hibernate/engine/spi/ActionQueue.java

```diff
@@ -105,7 +105,7 @@
 	 */
 	private static final LinkedHashMap<Class<? extends Executable>,ListProvider> EXECUTABLE_LISTS_MAP;
 	static {
-		EXECUTABLE_LISTS_MAP = new LinkedHashMap<Class<? extends Executable>,ListProvider>( 8 );
+		EXECUTABLE_LISTS_MAP = new LinkedHashMap<Class<? extends Executable>,ListProvider>( CollectionHelper.determineProperSizing( 8 ) );
```

Comment:
Given that `LinkedHashMap` extends from `HashMap`, the following common usage:
```
new LinkedHashMap<>( 8 );
```
almost always end up with resizing for load factor is 0.75 by default, not 1. So in this specific scenario, the map will be resized when the 7th class is inserted into the map, which defeats the purpose of the constructor parameter.

A common misuse, just similar to the case for `HashMap`. Only `ConcurrentHashMap`'s constructor take load factor into account.

Reply:
hi @NathanQingyangXu , I know about map resizes - but I still don't see the point.

First off, this is a static constant which is initialized only once per classloader so it's not really worth your time, nor mine, to try avoid a single collection resize.

Also this is the maintenance branch, on which each change needs to then be ported to 6 and it slows down the 6 development - which we can't have :) Let's please do only important fixes or changes we actually need: such minor improvements are not healthy for the project as they steal vital focus from other things.

But even if I don't think we should apply this, I'm happy to discuss this case in more detail as it's interesting. I believe 8 is a reasonable estimate, it's not possible to have this exact. It's a map, so keys are going to be hashed and therefore there are likely collisions, leading to some elements to share a bucket: that's why the constructor applies a load factor.

You can debug this and verify that most of the times with 8 there is either 1 or 2 resizings that need to happen: one is always going to be needed as the buckets are lazily allocated.
It's "most of the times" and not "always" as the key is a Class, and its hashcode will depend on its identity, which might be different if you run this different times. Working hard on making this resize only once in all cases would come at a memory penalty for the cases in which a single resize was good enough, so it's not a silver bullet either.

IMO if you want to really optimise this code path that's great, but I think there's potential here to not use a LinkedHashMap at all, benefit from immutability, and the change would belong in 6. At very least since this is a constant you could replace the 8 with a slightly larger value; e.g. using "10" would make a resize need more unlikely - but still not guaranteed either.

Reply:
After second-thought, I reverted back this code change for we have solve it in v6 in systematic way. No need to bring about merging conflict furthermore.